include/openenclave/bits/report.h

```diff
@@ -93,16 +97,40 @@ typedef struct _oe_identity
     uint64_t attributes;
 
     /** The unique ID for the enclave.
-      * For SGX enclaves, this is the MRENCLAVE value */
+      * For SGX enclaves, this is the MRENCLAVE value.
+      * For CyReS TAs, this is a hash of the TA image. */
     uint8_t unique_id[OE_UNIQUE_ID_SIZE];
 
     /** The signer ID for the enclave.
-      * For SGX enclaves, this is the MRSIGNER value */
+      * For SGX enclaves, this is the MRSIGNER value.
+      * For CyReS TAs, this is a hash of the public key that signed the TA image. */
     uint8_t signer_id[OE_SIGNER_ID_SIZE];
 
     /** The Product ID for the enclave.
      * For SGX enclaves, this is the ISVPRODID value. */
     uint8_t product_id[OE_PRODUCT_ID_SIZE];
+
+    /** The unique ID for the TA loader.
+     * For CyReS TAs, this is a combined hash of PPA image OP TEE image. */
+    uint8_t loader_unique_id[OE_UNIQUE_ID_SIZE];
```

Comment:
This structure seems geared towards what we have today (SPL->UBOOT->OPTEE->TA). But really, a “CyRes” report can have any number of intermediate measurements. The chain maybe smaller or larger depending on how the device boots. We will always have unique “device ID” and a leaf “TA ID”, the intermediate measurements may or may not be of interest.

I think we should re-think this structure a bit.

Reply:
Simplified for now as per offline conversation.